docs/develop/contracts/sandbox.md

```diff
@@ -0,0 +1,370 @@
+  });
+  assert.equal(bob_message, null);
+
+  // 4
+  await bobUseContract.set_status({ args: { message: "world" } });
+  bob_message = await bobUseContract.get_status({
+    account_id: "bob.test.near",
+  });
+  assert.equal(bob_message, "world");
+  alice_message = await aliceUseContract.get_status({
+    account_id: "alice.test.near",
+  });
+  assert.equal(alice_message, "hello");
+}
+
+test();
+```
+
+The test itself is very straightfoward. It does:
+
+1. Create testing accounts and deploy contract.
+2. Alice sign the transaction to set status, and get status to see set status works.
+3. Get Bob's status and that should be null because Bob has not set status.
+4. Bob set status and get status, should show Bob's status changed and not affect Alice status.
+
+> Most of the code above is boilerplate setup code to setup NEAR API, key pairs, testing accounts and deploy the contract. We're working on a near-cli `near test` command to do these setup code so you can focus on writing only `test()` for this kind of test.
+
+## Sandbox-only Features for Testing
+
+If you only use above test script that only uses standard NEAR rpcs, your tests can also be executed on a testnet node. Just replace the network ID, node url, key path and account names in above script and rerun. There's also some additional Sandbox-only features that make certain tests easier. We'll review some examples in this section.
+
+### Patch State on the Fly
+
+You can add or modify any contract state, contract code, account or access key during the test with `sandbox_patch_state` RPC.
+
+For arbitrary mutation on contract state, you cannot do it via transactions since transaction can only include contract calls that mutate state in contract programmed way. For example, for a NFT contract, you can do some operation with your owned NFTs but you cannot manipulate other people owned NFTs because smart contract has coded with checks to reject that. This is the expected behavior of the NFT contract. However you may want to change other people's NFT for test setup. This is called "arbitary mutation on contract state" and can be done by `sandbox_patch_state` RPC or stop the node, dump state as genesis, edit genesis and restart the node. The later approach is more complicated to do and also cannot be done without restart the node.
+
+For patch contract code, account or access key, you can add them with normal deploy contract, create account or add key actions in transaction, but that's also limited to your account or sub-account. `sandbox_patch_state` RPC does not have this restriction.
+
+Let's see an example of how patch_state would help in a test. Assume you want to mock the real state of a mainnet, where `alice.near` has set a status, and you want to retrieve that message. Above script doesn't work out of box, because your master account is `test.near` and you can only
+create account of `alice.test.near`, not `alice.near`. Patch state can solve this problem.
```

Comment:
```suggestion
Let's explore an example of how `patch_state` would help in a test. Assume you want to mock the real state of `mainnet` where `alice.near` has set a status and you want to retrieve that status message. The above script doesn't work out of box because your master account is `test.near` and you can only create an account of `alice.test.near` not `alice.near`. Patch state can solve this problem.
```